wordpress-mobile/WordPress-Android
# Edit post model provider

This PR introduces an `EditPostModelProvider` which encapsulates the edited post in the EditPostActivity. It also wraps all the setters in order to expose a `LiveData` endpoint the other parts can listen to. Right now the update calls are very messy. This also allows us to not use the casted activity hook and get current post from there. 

This solves one of the issues of the `EditPostActivity` - by not mutating the original object we could listen to the changes and introduce UI updates that are not called manually. Right now this is a great risk of missing an UI update when updating the Post. 

Let me know if this is the right direction! 

To test:
* Test that the edit post activity, settings and the publish settings work as expected. 

Update release notes:
